packages/vsx-registry/src/browser/vsx-extensions-contribution.ts

```diff
@@ -107,4 +156,33 @@ export class VSXExtensionsContribution extends AbstractViewContribution<VSXExten
         }
         return false;
     }
+
+    /**
+     * Installs a local .vsix file after prompting the `Open File` dialog. Resolves to the URI of the file.
+     */
+    protected async installFromVSIX(): Promise<URI | undefined> {
+        const props: OpenFileDialogProps = {
+            title: VSXExtensionsCommands.INSTALL_FROM_VSIX.dialogLabel,
+            openLabel: 'Install',
+            filters: { 'VSIX Extensions (*.vsix)': ['vsix'] },
+            canSelectMany: false
+        };
+        const extensionUri = await this.fileDialogService.showOpenDialog(props);
+        if (extensionUri) {
+            if (extensionUri.toString().endsWith('.vsix')) {
```

Comment:
Should we deal with the `path` instead?

```suggestion
            if (extensionUri.path.ext === '.vsix') {
```